libs/traits/src/liquidity_pools.rs

```diff
@@ -48,6 +50,37 @@ pub trait Router {
 	fn send(&self, sender: Self::Sender, message: Vec<u8>) -> DispatchResultWithPostInfo;
 }
 
+/// The behavior of an entity that can send messages
+pub trait MessageSender {
+	/// The originator of the message to be sent
+	type Origin;
+
+	///The destination of the message
+	type Destination;
+
+	/// Sends a message for origin to destination
+	fn send(
+		origin: Self::Origin,
+		destination: Self::Destination,
+		message: Vec<u8>,
+	) -> DispatchResultWithPostInfo;
+}
+
+/// The behavior of an entity that can receive messages
+pub trait MessageReceiver {
+	/// The maximum lenght for a message the implementor is able to receive.
+	type MaxEncodedLen: Get<u32>;
+
+	/// The originator of the message to be sent
+	type Origin;
+
+	/// Sends a message for origin to destination
+	fn receive(
+		origin: Self::Origin,
+		message: BoundedVec<u8, Self::MaxEncodedLen>,
+	) -> DispatchResult;
+}
```

Comment:
@Cosmin, this will be the new interface. Basically, the routers are no longer instances. They are pallet, but each pallet can have different configurations in they want.

I'm not fully sure yet, but my intention is that the Self::Origin will be a tuple `(RouterId, Domain)` or similar. Still thinking about this, but somehow though the receive()/send() method you could determine the router.

Hopefully, it will require few changes in the gateway.

Also the router storage and configuration could be removed from the gateway.

Reply:
> Also the router storage and configuration could be removed from the gateway.

OK, but then we need a way to somehow still "add" the routers to the gateway. Maybe add a tuple of all the available router pallets to the gateway config or something.

Reply:
Yeah, the way would be by adding a list of router ids

Reply:
And from the gateway perspective, only router ids exist

Reply:
OK, but then how would the gateway call router X to send an outbound message?

Reply:
`send(router_id, domain, message)` something like that